addons/website/static/src/snippets/s_process_steps/options.js

```diff
@@ -0,0 +1,56 @@
+odoo.define('website.s_process_steps_options', function (require) {
+    'use strict';
+    const options = require('web_editor.snippets.options');
+
+    options.registry.ConnectorChoice = options.Class.extend({
+        /**
+         * @override
+         */
+        start: function () {
+            if (!this.$target[0].dataset.currentConnector) {
+                this.$target[0].dataset.currentConnector = "line";
+            }
+            return this._super(...arguments);
+        },
+
+        //--------------------------------------------------------------------------
+        // Options
+        //--------------------------------------------------------------------------
+
+        /**
+         * Change connector type
+         */
+        changeConnector: function (previewMode, widgetValue, params) {
+            // We need to find connectors each time because user could have added column thus connectors.
+            const $rightConnectors = this.$('.s_process_step_line_right');
+            const $leftConnectors = this.$('.s_process_step_line_left');
+            const oldConnector = this.$target[0].dataset.currentConnector;
+            this.$target[0].dataset.currentConnector = widgetValue;
```

Comment:
In fact, I would not use that `currentConnector` at all, this is an attribute which seems redundant and useless. The `_computeWidgetState` can find if it's line or arrow without that attribute.